kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/cloudconnection/factory/CloudConnectionManagerFactory.java

```diff
@@ -0,0 +1,112 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Eurotech and/or its affiliates
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.cloudconnection.factory;
+
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.cloudconnection.CloudConnectionManager;
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * A {@link CloudConnectionManagerFactory} represents an OSGi Declarative Service Component
```

Comment:
"OSGi DS Component" is an internal detail. I think this should not be a part of API. You could simply write - "A {@link CloudConnectionManagerFactory} is responsible for registering {@link CloudConnectionManager} ..."